samples/tfm_integration/tfm_secure_partition/dummy_partition/dummy_partition.c

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <psa/crypto.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include "tfm_secure_api.h"
+#include "tfm_api.h"
+
+static psa_status_t tfm_dp_init(void)
+{
+	return TFM_SUCCESS;
+}
+
+#define NUM_SECRETS 5
+
+struct dp_secret {
+	uint8_t secret[16];
+};
+
+struct dp_secret secrets[NUM_SECRETS] = {
+	{{0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15}},
+	{{1,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15}},
+	{{2,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15}},
+	{{3,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15}},
+	{{4,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15}},
+};
+
+typedef void (*psa_write_callback_t)(void *handle, uint8_t *digest,
+				uint32_t digest_size);
+
+static psa_status_t tfm_dp_secret_digest(uint32_t secret_index,
+			size_t digest_size, size_t *p_digest_size,
+			psa_write_callback_t callback, void *handle)
+{
+	uint8_t digest[32];
+	psa_status_t status;
+
+	/* Check that secret_index is valid. */
+	if (secret_index >= NUM_SECRETS) {
+		return PSA_ERROR_INVALID_ARGUMENT;
+	}
+
+	/* Check that digest_size is valid. */
+	if (digest_size != sizeof(digest)) {
+		return PSA_ERROR_INVALID_ARGUMENT;
+	}
+
+	status = psa_hash_compute(PSA_ALG_SHA_256, secrets[secret_index].secret,
+				sizeof(secrets[secret_index].secret), digest,
+				digest_size, p_digest_size);
+
+	if (status != PSA_SUCCESS) {
+		return status;
+	}
+	if (*p_digest_size != digest_size){
+		return PSA_ERROR_PROGRAMMER_ERROR;
+	}
+
+	callback(handle, digest, digest_size);
+
+	return PSA_SUCCESS;
+}
+
+#ifndef TFM_PSA_API
+
+#include "tfm_memory_utils.h"
+
+/*
+ * \brief Indicates whether DP has been initialised.
+ */
+static bool dp_is_init = false;
+
+/*
+ * \brief Initialises DP, if not already initialised.
+ *
+ * \note In library mode, initialisation is delayed until the first secure
+ *	   function call, as calls to the Crypto service are required for
+ *	   initialisation.
+ *
+ * \return PSA_SUCCESS if DP is initialised, PSA_ERROR_GENERIC_ERROR
+ *		 otherwise.
+ */
+static psa_status_t dp_check_init(void)
+{
+	if (!dp_is_init) {
+		if (tfm_dp_init() != PSA_SUCCESS) {
+			return PSA_ERROR_GENERIC_ERROR;
+		}
+		dp_is_init = true;
+	}
+
+	return PSA_SUCCESS;
+}
```

Comment:
Can we remove this also?

Either I am reading the code wrong, or at some point this was necessary and then it got reworked to not be needed.